# NOTE: Do not use 1e-6 format because this is reqognized as a string

# Framework Settings
pub_on_prop: false
use_ros_time_now: true
verbose: false
verbose_out_of_order: true
discard_ooo_prop_meas: true
cov_debug: false
pub_cov: false
buffer_size: 2000

# Ros Settings
use_tcpnodelay: true
bypass_init_service: false

pub_cb_buffer_size: 1
sub_imu_cb_buffer_size: 200
sub_sensor_cb_buffer_size: 1

# Node specific settings
use_common_gps_reference: true

# Sensor Settings
# gyro_rate_noise: 0.0023089
# gyro_bias_noise: 0.0012655
# acc_noise: 0.038932
# acc_bias_noise: 0.057518
gyro_rate_noise: 7.9e-5
gyro_bias_noise: 1e-6
acc_noise: 1.714e-3
acc_bias_noise: 4.7e-5

# Core state covariance p,v,q,bw,ba
core_init_cov_p: [0.5,0.5,0.5]
core_init_cov_v: [0.3,0.3,0.3]
core_init_cov_q: [0.1218,0.1218,0.1218] # 20degree
core_init_cov_bw: [0.0076,0.0076,0.0076]
core_init_cov_ba: [0.01,0.01,0.01]

# Pose1 Settings
pose1_pos_meas_noise: [0.01,0.01,0.01]
pose1_att_meas_noise: [0.01,0.01,0.01]
pose1_cal_p_ip: [0,0,0]
pose1_cal_q_ip: [1,0,0,0]
pose1_state_init_cov: [0.00001,0.00001,0.00001,0.00001,0.00001,0.00001]

# Pose2 Settings
pose2_pos_meas_noise: [0.0,0.0,0.0]
pose2_att_meas_noise: [0,0,0]
pose2_cal_p_ip: [0,0,0]
pose2_cal_q_ip: [1,0,0,0]
pose2_state_init_cov: [0.00001,0.00001,0.00001,0.00001,0.00001,0.00001]

# GPS Settings
gps_pos_meas_noise: [1.0,1.0,1.0]
gps_vel_meas_noise: [2,2,2]
gps_cal_ig: [0,0,0.21]
gps_state_init_cov: [0.001,0.001,0.001]

# Baro Settings
baro_meas_noise: [1.0]
baro_cal_ip: [0,0,0]
baro_state_init_cov: [0.00001,0.00001,0.00001]
